diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdc66496a2..7a17a3a740 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1250,7 +1250,7 @@ elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "FREEBSD")
   set(SWIFT_HOST_VARIANT "freebsd" CACHE STRING
       "Deployment OS for Swift host tools (the compiler) [freebsd].")
 
-  configure_sdk_unix("FreeBSD" "${SWIFT_HOST_VARIANT_ARCH}")
+  configure_sdk_unix("FreeBSD" "aarch64")
   set(SWIFT_PRIMARY_VARIANT_SDK_default  "${SWIFT_HOST_VARIANT_SDK}")
   set(SWIFT_PRIMARY_VARIANT_ARCH_default "${SWIFT_HOST_VARIANT_ARCH}")
 
diff --git a/cmake/modules/Libdispatch.cmake b/cmake/modules/Libdispatch.cmake
index c8977e42d4..f29d8c6cf2 100644
--- a/cmake/modules/Libdispatch.cmake
+++ b/cmake/modules/Libdispatch.cmake
@@ -75,6 +75,10 @@ foreach(sdk ${DISPATCH_SDKS})
 
     if(sdk MATCHES WINDOWS)
       set(SWIFT_LIBDISPATCH_COMPILER_TRIPLE_CMAKE_ARGS -DCMAKE_C_COMPILER_TARGET=${SWIFT_SDK_WINDOWS_ARCH_${arch}_TRIPLE};-DCMAKE_CXX_COMPILER_TARGET=${SWIFT_SDK_WINDOWS_ARCH_${arch}_TRIPLE})
+    elseif(sdk MATCHES FREEBSD)
+      set(SWIFT_LIBDISPATCH_COMPILER_TRIPLE_CMAKE_ARGS -DCMAKE_C_COMPILER_TARGET=${SWIFT_SDK_FREEBSD_ARCH_${arch}_TRIPLE};-DCMAKE_CXX_COMPILER_TARGET=${SWIFT_SDK_FREEBSD_ARCH_${arch}_TRIPLE};-DCMAKE_SYSROOT=${SWIFT_SDK_FREEBSD_ARCH_${arch}_PATH};-DCMAKE_SYSTEM_NAME=FreeBSD)
+    else()
+      set(SWIFT_LIBDISPATCH_COMPILER_TRIPLE_CMAKE_ARGS)
     endif()
 
     if("${sdk}" STREQUAL "ANDROID")
