diff --git a/Package.swift b/Package.swift
index aad0c1c..c10f2ce 100644
--- a/Package.swift
+++ b/Package.swift
@@ -113,7 +113,7 @@ for target in package.targets {
     case .regular, .test, .executable:
         var settings = target.swiftSettings ?? []
         // https://github.com/swiftlang/swift-evolution/blob/main/proposals/0444-member-import-visibility.md
-        settings.append(.enableUpcomingFeature("MemberImportVisibility"))
+        // [freebsd-port] disabled: settings.append(.enableUpcomingFeature("MemberImportVisibility"))
         target.swiftSettings = settings
     case .macro, .plugin, .system, .binary:
         ()  // not applicable
diff --git a/Sources/CAsyncHTTPClient/CAsyncHTTPClient.c b/Sources/CAsyncHTTPClient/CAsyncHTTPClient.c
index 6342da8..ab52415 100644
--- a/Sources/CAsyncHTTPClient/CAsyncHTTPClient.c
+++ b/Sources/CAsyncHTTPClient/CAsyncHTTPClient.c
@@ -14,6 +14,8 @@
 
 #if __APPLE__
     #include <xlocale.h>
+#elif defined(__FreeBSD__)
+    #include <xlocale.h>
 #elif __linux__
     #include <locale.h>
 #endif
