diff --git a/Sources/Subprocess/Thread.swift b/Sources/Subprocess/Thread.swift
index 5e51b34..b450475 100644
--- a/Sources/Subprocess/Thread.swift
+++ b/Sources/Subprocess/Thread.swift
@@ -33,6 +33,10 @@ import Synchronization
 private typealias MutexType = CRITICAL_SECTION
 private typealias ConditionType = CONDITION_VARIABLE
 private typealias ThreadType = HANDLE
+#elseif os(FreeBSD)
+private typealias MutexType = pthread_mutex_t?
+private typealias ConditionType = pthread_cond_t?
+private typealias ThreadType = pthread_t
 #else
 private typealias MutexType = pthread_mutex_t
 private typealias ConditionType = pthread_cond_t
@@ -311,7 +315,7 @@ internal func pthread_create(
         (Unmanaged<AnyObject>.fromOpaque(context!).takeRetainedValue() as! Context).body()
         return context
     }
-    #if canImport(Glibc) || canImport(Bionic)
+    #if (canImport(Glibc) || canImport(Bionic)) && !os(FreeBSD)
     var thread = pthread_t()
     #else
     var thread: pthread_t?
@@ -325,7 +329,7 @@ internal func pthread_create(
     if rc != 0 {
         throw SubprocessError.UnderlyingError(rawValue: rc)
     }
-    #if canImport(Glibc) || canImport(Bionic)
+    #if (canImport(Glibc) || canImport(Bionic)) && !os(FreeBSD)
     return thread
     #else
     return thread!
