diff --git a/Package.swift b/Package.swift
index b77b90b..56316db 100644
--- a/Package.swift
+++ b/Package.swift
@@ -347,7 +347,7 @@ for target in package.targets {
     case .regular, .test, .executable:
         var settings = target.swiftSettings ?? []
         // https://github.com/swiftlang/swift-evolution/blob/main/proposals/0444-member-import-visibility.md
-        settings.append(.enableUpcomingFeature("MemberImportVisibility"))
+        // [freebsd-port] disabled: settings.append(.enableUpcomingFeature("MemberImportVisibility"))
         target.swiftSettings = settings
     case .macro, .plugin, .system, .binary:
         ()  // not applicable
